src/main/scala/intent/core/expectations/contain.scala

```diff
@@ -76,3 +76,40 @@ class ArrayContainExpectation[T](expect: Expect[Array[T]], expected: T)(
   def evaluate(): Future[ExpectationResult] =
     val actual = expect.evaluate()
     evalToContain(actual, expected, expect, "Array")
+
+class MapContainExpectation[K, V](expect: Expect[Map[K, V]], expected: Seq[Tuple2[K, V]])(
+  given
+    eqq: Eq[V],
+    fmt: Formatter[V]
+) extends Expectation with
+
+  def evaluate(): Future[ExpectationResult] =
+    val actual = expect.evaluate()
+    var failingKeys = Seq[Tuple2[K, V]]()       // When the key failing (missing or present when negated)
+    var failingValues = Seq[Tuple3[K, V, V]]()  // When the key is OK but the value does not match
+
+    expected.foreach { expectedPair =>
+      actual.get(expectedPair._1) match {
+        case None if expect.isNegated =>
+          () // OK = NOOP
+        case None if !expect.isNegated =>
```

Comment:
```suggestion
        case None =>
```

The condition is redundant since it's just a negation of the one on the previous line.